### PR TITLE
[COE] Refactoring status components

### PR DESCRIPTION
**Pull request by jerekshoe**
## Description
Now that COE is getting closer to final release, it's a good time to go through and refactor the contents. This PR is to refactor the components related to the status box that shows up on the form and the status apps.

These changes include:
* Remove 'Coe'/'COE' from component names
* Add `referenceNumber` prop to relevant components
* Rename the prop `applicationCreateDate` -> `requestDate`
* Moved `form/containers/introduction-content/COEStatuses` -> `form/components/statuses`
* Created a shared component called `StatusBox` to encapsulate the logic for rendering the alert at the beginning of the content
* Created a shared `ReviewAndDownload` component, to handle the 4 instances of the same content across different screens

## Original issue(s)
department-of-veterans-affairs/va.gov-team#36821

## Acceptance criteria
- [x] COE Status Components have been refactored

### CODE REVIEW

File: src/applications/lgy/coe/form/containers/IntroductionPage.jsx [hunk 1]
```diff
@@ -52,13 +54,6 @@ const IntroductionPage = ({ coe, downloadUrl, loggedIn, route, status }) => {
   );
 };
 
-IntroductionPage.propTypes = {
```
Comment (jerekshoe) [inline src/applications/lgy/coe/form/containers/IntroductionPage.jsx:55]: propTypes showed up in the code twice, so I removed this one and left the other

File: src/applications/lgy/coe/status/components/statuses/Ineligible.jsx [hunk 2]
```diff
@@ -2,32 +2,21 @@ import React from 'react';
...
-        </p>
-        <a href="/housing-assistance/home-loans/">
-          Learn more about VA-backed home loans
-        </a>
-      </va-alert>
+      <StatusBox.Ineligible />
       <h2>Can I get a VA home loan COE?</h2>
       <p>
         You may be able to get a COE if you didn’t receive a dishonorable
-        discharge and you meet the minimum active-duty service requirement based
```
Comment (jerekshoe) [inline src/applications/lgy/coe/status/components/statuses/Ineligible.jsx:29]: Removed `active-duty` here to align with mockups

File: src/applications/lgy/coe/status/components/statuses/Available.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      </p>
+      <ul>
+        <li>
+          Make changes to your COE (correct an error or update your
+          information), <strong>or</strong>
+        </li>
+        <li>Request a restoration of entitlement</li>
+      </ul>
+      <a href={introUrl}>
+        Make changes to your COE online by filling out VA Form 26-1880
```
Comment (jerekshoe) [inline src/applications/lgy/coe/status/components/statuses/Available.jsx:30]: The word `only` was replaced by `online` here, to align with content

File: src/applications/lgy/coe/shared/components/StatusBox/statuses/Available.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  <va-alert status="info">
+    <h2 slot="headline">You already have a COE</h2>
+    <div>
+      <p>You requested a COE on: {formatDate(requestDate)}</p>
+      <p>
+        You have a COE available so you don’t need to fill out a request. You
+        can review the details about your COE status or download your COE now.
+        <br />
+        <a href={statusUrl}>
+          Go to your VA home loan COE page to review the details of your COE
```
Comment (jerekshoe) [inline src/applications/lgy/coe/shared/components/StatusBox/statuses/Available.jsx:16]: removed the word `status` from the end of this text to align with mockups

File: src/applications/lgy/coe/shared/components/StatusBox/statuses/Pending.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    return 'You’ll need to upload the documents listed on this page. Then we can make a decision on your COE request.';
+  }
+  return 'If you qualify for a Certificate of Eligibility, we’ll notify you by email to let you know how to get your COE.';
+};
+
+const getLinkText = status =>
+  status === 'pending-upload'
+    ? 'Go to your VA home loan COE page to upload documents'
+    : 'Go to your VA home loan COE page to review the details of your COE';
```
Comment (jerekshoe) [inline src/applications/lgy/coe/shared/components/StatusBox/statuses/Pending.jsx:27]: Removed the word `status` from the end of this text to align with mockups

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by Mottie — APPROVED** [review]

**Review by cohnjesse — APPROVED** [review]
This looks great, really tidy refactor

**Pull request merged**